backend/api/messages.py

```
@@ -0,0 +1,468 @@
+    # Create notification for the recipient
+    if current_user.role == 'professor' and recipient.role == 'student':
+        create_professor_message_notification(
+            db=db,
+            professor=current_user,
+            student=recipient,
+            message_subject=message_data.subject,
+            cohort_id=message_data.cohort_id
+        )
+    elif current_user.role == 'student' and recipient.role == 'professor':
+        # For new messages from students to professors, use a different notification type
+        notification_service.create_notification(
```

Ruff (0.13.1)

Undefined name `notification_service`

(F821)